File: spec/javascript/test_bits_enum.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('BitsEnum', 'src/fixed_struct.bin', function(r, BitsEnum) {
  // 50 41 (4 + 8 + 1) = 0101|0000 0100|0|001
  assert.equal(r.one, BitsEnum.Animal.PLATYPUS);
  assert.equal(r.two, BitsEnum.Animal.HORSE);
  assert.equal(r.three, BitsEnum.Animal.CAT);
});

File: spec/java/src/io/kaitai/struct/spec/TestBitsEnum.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.BitsEnum;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestBitsEnum extends CommonSpec {
    @Test
    public void testBitsEnum() throws Exception {
        BitsEnum r = BitsEnum.fromFile(SRC_DIR + "fixed_struct.bin");

        // 50 41 (4 + 8 + 1) = 0101|0000 0100|0|001
        assertEquals(r.one(), BitsEnum.Animal.PLATYPUS);
        assertEquals(r.two(), BitsEnum.Animal.HORSE);
        assertEquals(r.three(), BitsEnum.Animal.CAT);
    }
}
